src/Microsoft.Azure.SignalR.Common/Utilities/ServiceMessageHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography;
+using Microsoft.Azure.SignalR.Protocol;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal static class ServiceMessageHelper
+    {
+        private static readonly RNGCryptoServiceProvider _keyGenerator = new RNGCryptoServiceProvider();
+        private static readonly char[] _padding = { '=' };
+
+        // todo: generate a shorter ID
+        public static string GenerateMessageId()
+        {
+            // 128 bit buffer / 8 bits per byte = 16 bytes
+            Span<byte> buffer = stackalloc byte[16];
+            var bufferArray = buffer.ToArray();
+            _keyGenerator.GetBytes(bufferArray);
```

Comment:
should we use a prefix + counter as a message id?
e.g.: `prefix + Interlocked.Increase(ref count).ToString("x")`

Reply:
Sounds good! Then we have the message order .

Reply:
Then prefix is a 1:1 ID to server connection, but shorter than server connection ID.
The message ID will be: `Prefix.MessageIndexHex`